### ISSUE TITLE
Tooltips not shown for F# solution

### ISSUE THREAD

**Issue by 7sharp9** [mono/monodevelop#7254]
The following error is shown in the log on trying to display a tooltip:

```
WARNING [2019-02-26 21:29:36Z]: TooltipProvider: Error: System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: Operation not supported in background thread
  at MonoDevelop.Core.Runtime.AssertMainThread () [0x00012] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:450 
  at Mono.TextEditor.TextEditorData.GetChunks (Mono.TextEditor.DocumentLine line, System.Int32 offset, System.Int32 length, System.Threading.CancellationToken cancellationToken) [0x00037] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/TextEditorData.cs:500 
  at Mono.TextEditor.TextEditorData.GetMarkupAsync (System.Int32 offset, System.Int32 length, System.Boolean removeIndent, System.Boolean useColors, System.Boolean replaceTabs, System.Boolean fitIdeStyle, System.Threading.CancellationToken cancellationToken) [0x00180] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/TextEditorData.cs:449 
   --- End of inner exception stack trace ---
  at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00013] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-06/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2054 
  at System.Threading.Tasks.Task.Wait (System.Int32 millisecondsTimeout, System.Threading.CancellationToken cancellationToken) [0x00043] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-06/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2784 
  at System.Threading.Tasks.Task.Wait (System.Threading.CancellationToken cancellationToken) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-06/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2703 
  at MonoDevelop.Ide.TaskUtil.WaitAndGetResult[T] (System.Threading.Tasks.Task`1[TResult] task, System.Threading.CancellationToken cancellationToken) [0x00000] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/TaskUtil.cs:54 
  at Mono.TextEditor.TextEditorData.GetMarkup (System.Int32 offset, System.Int32 length, System.Boolean removeIndent, System.Boolean useColors, System.Boolean replaceTabs, System.Boolean fitIdeStyle) [0x00000] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/TextEditorData.cs:416 
  at MonoDevelop.FSharp.Highlight.syntaxHighlight (System.String s) [0x00025] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/external/fsharpbinding/MonoDevelop.FSharpBinding/FSharpSymbolHelper.fs:381 
  at <StartupCode$FSharpBinding>.$FSharpTooltipProvider.binder@1 (MonoDevelop.Ide.Editor.TextEditor editor, System.String lineStr, System.Int32 col, Microsoft.FSharp.Compiler.SourceCodeServices.FSharpSymbolUse _arg1, System.String _arg20, MonoDevelop.FSharp.Shared.XmlDoc _arg21, System.String _arg22) [0x00000] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/external/fsharpbinding/MonoDevelop.FSharpBinding/FSharpTooltipProvider.fs:74 
  at <StartupCode$FSharpBinding>.$FSharpTooltipProvider+tooltipComputation@70-6.Invoke (Microsoft.FSharp.Core.Unit unitVar) [0x00048] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/external/fsharpbinding/MonoDevelop.FSharpBinding/FSharpTooltipProvider.fs:0 
  at Microsoft.FSharp.Control.AsyncBuilderImpl+callA@522[b,a].Invoke (Microsoft.FSharp.Control.AsyncParams`1[T] args) [0x00051] in <5a7203e1904cf4daa7450383e103725a>:0 
---> (Inner Exception #0) System.InvalidOperationException: Operation not supported in background thread
  at MonoDevelop.Core.Runtime.AssertMainThread () [0x00012] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Core/Runtime.cs:450 
  at Mono.TextEditor.TextEditorData.GetChunks (Mono.TextEditor.DocumentLine line, System.Int32 offset, System.Int32 length, System.Threading.CancellationToken cancellationToken) [0x00037] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/TextEditorData.cs:500 
  at Mono.TextEditor.TextEditorData.GetMarkupAsync (System.Int32 offset, System.Int32 length, System.Boolean removeIndent, System.Boolean useColors, System.Boolean replaceTabs, System.Boolean fitIdeStyle, System.Threading.CancellationToken cancellationToken) [0x00180] in /Users/vsts/agent/2.147.1/work/1/s/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/TextEditorData.cs:449 <---
```

> VS bug [#802684](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/802684)

**Comment by 7sharp9** [comment]
Ive not included any repo as this occurs with pretty much every repo I tried, if you want a specific one:
https://github.com/jet/falanx/Falanx.Machinery/Prelude.fs
hover anything in that file, no tooltip, error in log.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-15] Comment by **xamarinc**



Fixed in version 8.1.0.63 (master)



Pull Request #7282 merged by: Lluis Sanchez  
 Author: mono  
 Commit: de1c2b731e56984b7004a956030b94a5fa003da7 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-16] Comment by **xamarinc**



Fixed in version 7.8.2.3 (release-7.8)



Pull Request #7283 merged by: Lluis Sanchez  
 Author: mono  
 Commit: 2c3605c5b1e708cde4e1b330fdc5c92f79fed611 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-23] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #7282 merged by: Lluis Sanchez  
 Author: mono  
 Commit: de1c2b731e56984b7004a956030b94a5fa003da7 (xamarin/monodevelop)
